### ISSUE TITLE
WYSIWYM (visual editor) controls duplicated on format change

### ISSUE THREAD

**Issue by drmowinckels** [rstudio/rstudio#12227]
---
Edit on 2023-10-11 by RStudio Dev Team:
A temporary workaround to resolve this is to close and reopen the file.
---

When changing format in the visual editor, a prompt to change the editor appears to follow the new format.
It then seems like the WYSIWYM controls are pasted next to the last over and over upon changing formats.
Leaving the ones visible not working, but the new ones hard to access. 

![image](https://user-images.githubusercontent.com/14014329/198004066-67ee5c83-3a0f-41b4-8256-17d901dab74f.png)

https://user-images.githubusercontent.com/14014329/198004762-a9944d6a-a58a-45d4-8119-1b24e78ec054.mov

RStudio: 
```
Spotted Wakerobin" Release (7872775e, 2022-07-22) for macOS
Mozilla/5.0 (Macintosh; Intel Mac OS X 12_6_0) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.10 Chrome/69.0.3497.128 Safari/537.36
```

```r
R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.6

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets 
[6] methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.2.0 tools_4.2.0   
```

**Comment by ronblum** [comment]
@drmowinckels Thank you for raising the issue! I can reproduce it in 2022.12.0-daily-286 on MacOS 13. In addition:
- Changing the format again adds another instance of the toolbar.
- Switching back and forth between source and visual mode doesn't resolve the problem.
- The only way to resolve this is to close and reopen the file.

We'll review this as we continue development of RStudio and fixing issues.

**Comment by ronblum** [comment]
See also @mine-cetinkaya-rundel's report at #13724.

**Comment by ronblum** [comment]
Verified fixed in RStudio Desktop 2024.04.0-daily+327 on MacOS 14.3.
